spring5/src/main/java/org/jdbi/v3/spring5/JdbiRepositoryRegistrar.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.spring5;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.annotation.AnnotatedGenericBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
+import org.springframework.core.type.AnnotationMetadata;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.StringUtils;
+
+/**
+ * This bean registers the bean definitions of all repositories.
+ * Interfaces found using the configuration of {@link EnableJdbiRepositories}
+ * and annotated with {@link JdbiRepository} will be registered.
+ */
+public class JdbiRepositoryRegistrar implements ImportBeanDefinitionRegistrar {
+
+    @Override
+    public void registerBeanDefinitions(AnnotationMetadata metadata, BeanDefinitionRegistry registry) {
+        Iterable<BeanDefinition> repositoryBeanDefinitions = resolveRepositoryBeanDefinitions(metadata);
+        for (BeanDefinition repositoryBeanDefinition : repositoryBeanDefinitions) {
+            AnnotationMetadata annotationMetadata = ((AnnotatedBeanDefinition) repositoryBeanDefinition).getMetadata();
+            registerJdbiRepositoryFactoryBean(registry, annotationMetadata);
+        }
+    }
+
+    private void registerJdbiRepositoryFactoryBean(BeanDefinitionRegistry registry, AnnotationMetadata annotationMetadata) {
+        String className = annotationMetadata.getClassName();
+        @SuppressWarnings("rawtypes")
+        Class clazz = ClassUtils.resolveClassName(className, null);
+
+        Map<String, Object> attributes = getAnnotationAttributes(JdbiRepository.class, annotationMetadata);
+        String jdbiQualifier = (String) attributes.get("jdbiQualifier");
```

Comment:
It's too bad you cannot use the annotation type itself here, like `anno.jdbiQualifier()`, but I guess this is how spring does this?